tee-worker/omni-executor/intent/executors/cross-chain/src/cross_chain.rs

```diff
@@ -96,21 +118,78 @@ impl<BinanceClient: BinanceApi, SolanaClient: SolanaClientTrait>
 		)
 		.await?;
 
-		// TODO: this should be abstracted away
-		let (payout_amount, payout_amount_u256) = self
-			.do_binance_swap(
+		let mut payout_amount =
+			self.estimate_payout(swap_order.from_asset.clone(), from_amount_decimal).await?;
+		let mut payout_amount_u256 = str_to_u256(&payout_amount, 18)?;
+
+		if instant {
+			Ok((
+				self.apply_gas_fee(&payout_amount, access_token, pumpx_config).await?,
+				self.accounting_contract_client.get_address().await.to_string(),
```

Comment:
I'm not sure that's correct

This address will be used as `address` field when calling `create_market_order_unsigned_tx` and it has to be an EOA (able to sign), but here it seems the accounting contract address :(

Reply:
should be resolved